l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -688,23 +736,39 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
     function depositLegacyErc20Bridge(
         address _prevMsgSender,
         address _l2Receiver,
-        address _l1Token,
+        address _l1Asset,
         uint256 _amount,
         uint256 _l2TxGasLimit,
         uint256 _l2TxGasPerPubdataByte,
         address _refundRecipient
     ) external payable override onlyLegacyBridge nonReentrant whenNotPaused returns (bytes32 l2TxHash) {
         require(l2BridgeAddress[ERA_CHAIN_ID] != address(0), "ShB b. n dep");
-        require(_l1Token != L1_WETH_TOKEN, "ShB: WETH deposit not supported 2");
+        require(_l1Asset != L1_WETH_TOKEN, "ShB: WETH deposit not supported 2");
 
-        // Note that funds have been transferred to this contract in the legacy ERC20 bridge.
-        if (!hyperbridgingEnabled[ERA_CHAIN_ID]) {
-            chainBalance[ERA_CHAIN_ID][_l1Token] += _amount;
-        }
+        bytes32 _assetId;
+        bytes memory bridgeMintCalldata;
 
-        bytes memory l2TxCalldata = _getDepositL2Calldata(_prevMsgSender, _l2Receiver, _l1Token, _amount);
+        {
+            // Inner call to encode data to decrease local var numbers
+            _assetId = nativeTokenVault.getAssetId(_l1Asset);
+            if (nativeTokenVault.tokenAddress(_assetId) == address(0)) {
+                nativeTokenVault.registerToken(_l1Asset);
+            }
+            _transferAllowanceToNTV(_assetId, _amount, _prevMsgSender);
```

Comment:
I don't think we even need to do it, since no allowance will be charged in the subsequent transactions

On the contrary, we should update legacy bridge to deposit funds to the NTV instead of shared bridge.